k8s/tikv/README.md

```diff
@@ -0,0 +1,370 @@
+
+- Set the persistent disks size. The default disks size is "2Gi".
+
+```shell
+export DEFAULT_STORAGE_CLASS="standard" # provide your StorageClass name if not "standard"
+export TIKV_PERSISTENT_DISK_SIZE="2Gi"
+```
+
+Set up the image tag:
+
+It is advised to use a stable image reference, which you can find on:
+- [TiKV - Marketplace Container Registry](https://gcr.io/ccm-ops-test-adhoc/tikv5).
+For example:
+
+```shell
+export TIKV_TRACK=5.3
+export METRICS_EXPORTER_TAG=0.5
+```
+
+Configure the container images:
+
+```shell
+export IMAGE_TIKV=gcr.io/ccm-ops-test-adhoc/tikv5
+export IMAGE_METRICS_EXPORTER=k8s.gcr.io/prometheus-to-sd:${METRICS_EXPORTER_TAG}
+```
+
+By default, TiKV deployment has 3 replica, but you can choose to set the number of replicas.
+
+```shell
+export TIKV_REPLICAS=3
+```
+
+(Optional) Enable Stackdriver Metrics Exporter:
+
+> **NOTE:** Your GCP project must have Stackdriver enabled. If you are using a
+> non-GCP cluster, you cannot export metrics to Stackdriver.
+
+By default, the application does not export metrics to Stackdriver. To enable
+this option, change the value to `true`.
+
+```shell
+export METRICS_EXPORTER_ENABLED=false
+```
+
+#### Expand the manifest template
+
+Use `helm template` to expand the template. We recommend that you save the
+expanded manifest file for future updates to your app.
+
+```shell
+helm template "${APP_INSTANCE_NAME}" chart/tikv \
+    --namespace "${NAMESPACE}" \
+    --set tikv.image.repo="$IMAGE_TIKV" \
+    --set tikv.image.tag="$TIKV_TRACK" \
+    --set tikv.persistence.storageClass="${DEFAULT_STORAGE_CLASS}" \
+    --set tikv.persistence.size="${TIKV_PERSISTENT_DISK_SIZE}" \
+    --set tikv.replicas="${TIKV_REPLICAS:-1}" \
+    --set metrics.image="$IMAGE_METRICS_EXPORTER" \
+    --set metrics.exporter.enabled="$METRICS_EXPORTER_ENABLED" \
+    > "${APP_INSTANCE_NAME}_manifest.yaml"
+```
+
+#### Apply the manifest to your Kubernetes cluster
+
+Use `kubectl` to apply the manifest to your Kubernetes cluster:
+
+```shell
+kubectl apply -f "${APP_INSTANCE_NAME}_manifest.yaml" --namespace "${NAMESPACE}"
+```
+
+#### View the app in the Google Cloud Console
+
+To get the Cloud Console URL for your app, run the following command:
+
+```shell
+echo "https://console.cloud.google.com/kubernetes/application/${ZONE}/${CLUSTER}/${NAMESPACE}/${APP_INSTANCE_NAME}"
+```
+
+### Access TiKV through an external IP address
+
+By default, the application does not have an external IP address. To create an external IP address, run the following command:
+
+```
+kubectl patch svc "$APP_INSTANCE_NAME-pd" \
+  --namespace "$NAMESPACE" \
+  --patch '{"spec": {"type": "LoadBalancer"}}'
+```
+
+> **NOTE:** It might take some time for the external IP to be provisioned.
+
+### Access the TiKV's PD service
+
+If you run your TiKV cluster behind a LoadBalancer, you can get the external IP of the PD service using the following command:
+
+```shell
+PD_IP=$(kubectl get svc $APP_INSTANCE_NAME-pd \
```

Comment:
keep consitent var expansion: https://google.github.io/styleguide/shellguide.html#variable-expansion

Reply:
Here and everywhere in the file

Reply:
done